src/app/cards/cards-routing.module.ts

```diff
@@ -13,6 +14,10 @@ const routes: Routes = [
     component: CardsPageComponent,
     runGuardsAndResolvers: 'paramsOrQueryParamsChange'
   },
+  {
+    path: 'search',
```

Comment:
This needs to come before `:cardId` since otherwise it'll try to interpret the search as a card id.